travel-planner/travel-planner-ui/src/Components/SelectDirections/SelectDirections.jsx

```diff
@@ -0,0 +1,116 @@
+import "./SelectDirections.css";
+import React, { useState, useEffect } from "react";
+import axios from "axios";
+import Dropdown from "react-bootstrap/Dropdown";
+import Form from "react-bootstrap/Form";
+
+export default function SelectDirections({ PORT, mapId }) {
+	// get all directions from backend
+	const [directions, setDirections] = useState(null);
+  const [curDirections, setCurDirections] = useState(null);
+
+
+	const CustomToggle = React.forwardRef(({ children, onClick }, ref) => (
+		<a
+			href=""
+			ref={ref}
+			onClick={(e) => {
+				e.preventDefault();
+				onClick(e);
+			}}
+		>
+			{children}
+			&#x25bc;
+		</a>
+	));
+
+	const CustomMenu = React.forwardRef(
+		({ children, style, className, "aria-labelledby": labeledBy }, ref) => {
+      const [value, setValue] = useState("");
+
+			return (
+				<div
+					ref={ref}
+					style={style}
+					className={className}
+					aria-labelledby={labeledBy}
+				>
+					<Form.Control
+						autoFocus
+						className="mx-3 my-2 w-auto"
+						placeholder="Type to filter..."
+						onChange={(e) => setValue(e.target.value)}
+						value={value}
+					/>
+					<ul className="list-unstyled">
+						{React.Children.toArray(children).filter(
+							(child) =>
+								!value ||
+                child.props.children
+                  .join('')
+									.toLowerCase()
+									.includes(value.toLowerCase())
+						)}
+					</ul>
+				</div>
+			);
+		}
+	);
+
+	useEffect(async () => {
```

Comment:
On this issue (2.d) I mentioned how we integrate useEffect and async functions https://github.com/KimberlyTruong/tdd-student-store-starter/issues/2